### PR TITLE
Add user,account,profile datasource and token resource

### PR DESCRIPTION
**Pull request by displague**
FIXES:
* `linode_nodebalancer_config.check_passive` changes were not handled [GH-4] 

FEATURES:
* **New Resource** `linode_token` [[api]](https://developers.linode.com/api/v4#operation/createPersonalAccessToken)
* **New Data Resource** `linode_user` (username) [[api]](https://developers.linode.com/api/v4#operation/getUsers)
* **New Data Resource** `linode_account` [[api]](https://developers.linode.com/api/v4#operation/getAccount)
* **New Data Resource** `linode_profile` [[api]](https://developers.linode.com/api/v4#operation/getProfile)

### CODE REVIEW

File: linode/resource_linode_token.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+				Required:    true,
+				ForceNew:    true,
+			},
+			"expiry": &schema.Schema{
+				Type:             schema.TypeString,
+				Description:      "When this token will expire. Personal Access Tokens cannot be renewed, so after this time the token will be completely unusable and a new token will need to be generated. Tokens may be created with 'null' as their expiry and will never expire unless revoked.",
+				Optional:         true,
+				ValidateFunc:     validDateTime,
+				ForceNew:         true,
+				DiffSuppressFunc: equivalentDate,
```
Comment (cliedeman) [inline linode/resource_linode_token.go]: Did not know about this. Good to know

File: website/docs/d/user.html.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  Provides details about a Linode user.
+---
+
+# Data Source: user
+
+Provides information about a Linode user
+
+## Example Usage
+
+The following example shows how one might use this data source to define a Linode instance.
```
Comment (cliedeman) [inline website/docs/d/user.html.md]: instance -> user

File: website/docs/r/instance.html.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ Provides a Linode instance resource.  This can be used to create,
 modify, and delete Linodes. For more information, see [Getting Started with Linode](https://linode.com/docs/getting-started/)
 and [Linode APIv4 docs](https://developers.linode.com/api/v4#operation/createLinodeInstance).
 
-Linodes also support `[provisioning](/docs/provisioners/index.html).
+Linodes also support [provisioning](/docs/provisioners/index.html).
```
Comment (cliedeman) [inline website/docs/r/instance.html.md]: nit: supports

File: linode/data_source_linode_account.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+				Computed:    true,
+			},
+			"zip": {
+				Type:        schema.TypeString,
+				Description: "The zip code of this Account's billing address.",
+				Computed:    true,
+			},
+			"credit_card": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
```
[the remark this answers is not in the document]
Comment (displague) [reply linode/data_source_linode_account.go]: As far as I am aware, I have three options:

- **TypeSet**:  `linode_account.me.credit_card[12315154335124].last_four` the computed hash of a TypeSet makes it hard to access
- **TypeList**: `linode_account.me.credit_card[0].last_four` a list has the clunkiness of "why is this a list", but it has been the convention for making well-defined substructures accessible
- **TypeMap**: `linode_account.me.credit_card["last_four"]` a map is best used when you don't know what the keys will be and the values will be of the same type.

Another option would be to flatten this structure:
- *Flatten* `linode_account.me.credit_card_last_four` but this does not match the pattern one would expect using the Linode APIv4 reference as a guide. The design advice I have received has been to match the API as closely as possible.  Using TypeList of MaxItems 1 has been a predictable compromise.

File: linode/resource_linode_token.go (hunk 1 above)
Comment (displague) [reply linode/resource_linode_token.go]: I made up [`equivalentDate`](https://github.com/terraform-providers/terraform-provider-linode/pull/5/files/198e4b4e30e628ce9720cccb77d621a22b89c6eb#diff-0d95025d49ee76643538ff7d125f1495R58) and [`validDateTime`](https://github.com/terraform-providers/terraform-provider-linode/pull/5/files/198e4b4e30e628ce9720cccb77d621a22b89c6eb#diff-0d95025d49ee76643538ff7d125f1495R70).

I assume you mean  [DiffSuppressFunc](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#diffsuppressfunc).

**Pull request merged**